src/middlewared/middlewared/plugins/smb.py

```diff
@@ -98,6 +98,25 @@ def check_codec(encoding):
 
         return encodings
 
+    @private
+    async def store_ldap_admin_password(self):
+        """
+        This is required if the LDAP directory service is enabled. The ldap admin dn and
+        password are stored in private/secrets.tdb file.
+        """
+        ldap = await self.middleware.call('datastore.config', 'directoryservice.ldap')
+        password_stored = await self.middleware.call('cache.has_key', 'SMB_LDAP_BINDPW_SET')
```

Comment:
Shouldnt we pop SMB_LDAP_BINDPW_SET if ldap_bindpw changes?

Reply:
After I wrote this I thought of a few other edge cases where we might end up with secrets.tdb getting out of sync. For now (until we have a proper LDAP plugin) it's probably best to just reset the password every time we run the ctl script. It adds at most a second to the service start.